From e13eab438b401dbfdcf457b1eeb5646224d5fa47 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Tue, 10 Mar 2026 13:08:57 -0700
Subject: [PATCH 8/8] roctracer enable ASAN_PACKAGING for asan builds

Fixes a following error on ASAN builds.

33.2    [40/42] Linking HIP executable test/MatrixTranspose_ctest
33.2    FAILED: test/MatrixTranspose_ctest
...
33.2    ld.lld: error: undefined symbol: __asan_option_detect_stack_use_after_return
33.2    >>> referenced by MatrixTranspose.c:46 (test/MatrixTranspose.c:46)
33.2    >>>               test/CMakeFiles/MatrixTranspose_ctest.dir/MatrixTranspose.c.o:(SPRINT)
33.2    >>> referenced by MatrixTranspose.c:322 (test/MatrixTranspose.c:322)
33.2    >>>               test/CMakeFiles/MatrixTranspose_ctest.dir/MatrixTranspose.c.o:(activity_callback)
33.2    >>> referenced by MatrixTranspose.c:353 (test/MatrixTranspose.c:353)
33.2    >>>               test/CMakeFiles/MatrixTranspose_ctest.dir/MatrixTranspose.c.o:(init_tracing)
33.2
33.2    ld.lld: error: undefined symbol: __asan_stack_malloc_1
33.2    >>> referenced by MatrixTranspose.c:46 (test/MatrixTranspose.c:46)
33.2    >>>               test/CMakeFiles/MatrixTranspose_ctest.dir/MatrixTranspose.c.o:(SPRINT)
33.2    >>> referenced by MatrixTranspose.c:353 (test/MatrixTranspose.c:353)
33.2    >>>               test/CMakeFiles/MatrixTranspose_ctest.dir/MatrixTranspose.c.o:(init_tracing)

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 profiler/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/profiler/CMakeLists.txt b/profiler/CMakeLists.txt
index 83465f3..f20fc01 100644
--- a/profiler/CMakeLists.txt
+++ b/profiler/CMakeLists.txt
@@ -183,6 +183,7 @@ if(THEROCK_ENABLE_ROCPROFV3)
       amd-hip
     CMAKE_ARGS
       -DHIP_PLATFORM=amd
+      -DENABLE_ASAN_PACKAGING=${ENABLE_ASAN_PACKAGING}
     INTERFACE_INCLUDE_DIRS
       # All old clients of roctx64 expect to just be able to find its include
       # with no further qualification. Bad design, but also deprecated, so meh.
-- 
2.43.0

